cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersValidator.java

```diff
@@ -186,6 +189,39 @@ private static void validateIngress(Set<String> errors, int replicas, GenericKaf
         }
     }
 
+    /**
+     * Validates that ClusterIP type listener has the right host configurations
+     *
+     * @param errors    List where any found errors will be added
+     * @param replicas  Number of Kafka replicas
+     * @param listener  Listener which needs to be validated
+     */
+    private static void validateClusterIP(Set<String> errors, int replicas, GenericKafkaListener listener) {
+        if (listener.getConfiguration() != null)    {
+            GenericKafkaListenerConfiguration conf = listener.getConfiguration();
+
+            if (conf.getBrokers() != null) {
+                for (int i = 0; i < replicas; i++)  {
+                    final int id = i;
+                    GenericKafkaListenerConfigurationBroker broker = conf.getBrokers().stream().filter(b -> b.getBroker() == id).findFirst().orElse(null);
+
+                    if (broker == null || broker.getAdvertisedHost() == null) {
+                        errors.add("listener " + listener.getName() + " is missing a broker advertised host name for broker with ID " + i + " which is required for ClusterIP based listeners");
+                    }
+                    if (broker == null || broker.getAdvertisedPort() == null) {
+                        errors.add("listener " + listener.getName() + " is missing a broker advertised port for broker with ID " + i + " which is required for ClusterIP based listeners");
+                    }
+                }
+            } else {
+                errors.add("listener " + listener.getName() + " is missing a broker configuration with advertised host names which is required for ClusterIP based listeners");
+            }
+
+
+        } else {
+            errors.add("listener " + listener.getName() + " is missing a configuration with advertised host names which is required for ClusterIP based listeners");
+        }
```

Comment:
I don't think you should make the advertised host / ports required. Instead, you should update the `KafkaListenersReconciler` class and have it by defualt advertise the DNS names of the ClusterIP services / bootstrap service and add them to the TLS certificates. You can check how its done for example here: https://github.com/strimzi/strimzi-kafka-operator/blob/9978b91fa8c714df61b63b1ccd348b94871b9098/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenersReconciler.java#L404 or in the other methods in this class. That way, it will work out of the box and you can optionally override it.

Reply:
do you think is good idea to add ClusterIp listener type to ListenersUtils.internalListeners method? because when I say internalServicesReady method I dont think I need more with ClusterIP listeners. the other solution is to add a similar method (clusterIPServicesReady?) with the same content as internalServicesReady in the beginning

Reply:
I think you can do a new method for this listener type. That is what I would probably do. But the `internalServicesReady` and the other `*Ready` methods should give you the idea what would be in your new method.